source/grid/grid_tools.cc

```diff
@@ -2770,15 +2770,43 @@ namespace GridTools
 
 
 
+  namespace
+  {
+    template <int dim, int spacedim>
+    double diameter(const typename Triangulation<dim, spacedim>::cell_iterator &cell,
```

Comment:
is this exactly what `cell->diameter()` does, ie. look at vertices? If so, can we avoid somehow code duplication and join the two?